From ec3477947ae60275484a8ed87d2a379ab0da0455 Mon Sep 17 00:00:00 2001
From: Kerong Sui <ksui@codeaurora.org>
Date: Mon, 17 Feb 2014 16:12:48 +0800
Subject: [PATCH 104/176] mm-qcamera-app: take picture no matter focus is
 success or not

Now the snapshot will be hang when auto focus is not success, should
do the snapshot no matter the auto focus is success or not.

Change-Id: I9a7146a9f8b36ac5c996175a85f3e178745d7626
---
 QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c b/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c
index 3ac9eca..33bd30f 100644
--- a/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c
+++ b/QCamera2/stack/mm-camera-test/src/mm_qcamera_preview.c
@@ -81,7 +81,7 @@ static void mm_app_metadata_notify_cb(mm_camera_super_buf_t *bufs,
   if (pMetadata->is_focus_valid) {
     focus_data = (cam_auto_focus_data_t *)&(pMetadata->focus_data);
 
-    if (focus_data->focus_state == CAM_AF_FOCUSED) {
+    if (focus_data->focus_state == CAM_AF_FOCUSED || focus_data->focus_state == CAM_AF_NOT_FOCUSED) {
       CDBG_ERROR("%s: AutoFocus Done Call Back Received\n",__func__);
       mm_camera_app_done();
     }
-- 
1.8.3.1

